daemon/src/close_cfds.rs

```diff
@@ -0,0 +1,417 @@
+            taker_leverage,
+            n_contracts,
+            counterparty_network_identity,
+            role,
+            opening_fee,
+            initial_funding_rate,
+            initial_tx_fee_rate,
+            fee_account,
+            own_script_pubkey: None,
+            lock: None,
+            commit: None,
+            collaborative_settlement: None,
+            non_collaborative_settlement: None,
+            refund: None,
+            close_status: CloseStatus::NotClosed,
+            version: 0,
+        }
+    }
+
+    fn is_closed(&self) -> bool {
+        self.close_status != CloseStatus::NotClosed
+    }
+}
+
+impl CfdAggregate for Cfd {
+    type CtorArgs = ();
+
+    fn new(args: Self::CtorArgs, cfd: db::Cfd) -> Self {
+        Cfd::new(cfd)
+    }
+
+    fn apply(mut self, event: model::CfdEvent) -> Self {
+        use model::EventKind::*;
+        match event.event {
+            ContractSetupStarted => {}
+            ContractSetupCompleted { dlc } => {
+                let script_pubkey = dlc.lock.1.script_pubkey();
+                let OutPoint { txid, vout } = dlc
+                    .lock
+                    .0
+                    .outpoint(&script_pubkey)
+                    .expect("DLC to be in lock TX");
+
+                let txid = Txid::new(txid);
+                let dlc_vout = Vout::new(vout);
+
+                self.lock = Some(Lock { txid, dlc_vout });
+
+                self.own_script_pubkey = Some(dlc.script_pubkey_for(self.role));
+            }
+            ContractSetupFailed => {}
+            OfferRejected => {}
+            RolloverStarted => {}
+            RolloverAccepted => {}
+            RolloverRejected => {}
+            RolloverCompleted { dlc, funding_fee } => {
+                self.fee_account.add_funding_fee(funding_fee);
+            }
+            RolloverFailed => {}
+            CollaborativeSettlementStarted { proposal } => {}
+            CollaborativeSettlementProposalAccepted => {}
+            CollaborativeSettlementCompleted {
+                spend_tx,
+                script,
+                price,
+            } => {
+                let OutPoint { txid, vout } = spend_tx
+                    .outpoint(&script)
+                    .expect("spend script to be in spend tx");
+
+                let payout = &spend_tx.output[vout as usize];
+                let payout = Payout::new(Amount::from_sat(payout.value));
+
+                let txid = Txid::new(txid);
+                let vout = Vout::new(vout);
+
+                self.collaborative_settlement = Some(CollaborativeSettlement {
+                    txid,
+                    vout,
+                    payout,
+                    price,
+                })
+            }
+            CollaborativeSettlementRejected => {}
+            CollaborativeSettlementFailed => {}
+            LockConfirmed => {}
+            LockConfirmedAfterFinality => {}
+            CommitConfirmed => {}
```

Comment:
Mark the commit transaction as actually used here?

Reply:
Do we care about that in `projection`?